tensorboard/webapp/metrics/views/main_view/common_selectors_test.ts

```diff
@@ -430,4 +552,351 @@ describe('common selectors', () => {
+      ).toBeFalse();
+    });
+  });
+
+  describe('filterRunItemsByRegex', () => {
+    it('returns all runs when no regex is provided', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          '',
+          false
+        )
+      ).toEqual(runTableItems);
+    });
+
+    it('only returns runs matching regex', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'run',
+          false
+        )
+      ).toEqual(runTableItems);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'exp',
+          false
+        )
+      ).toEqual([]);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'exp',
+          true
+        )
+      ).toEqual(runTableItems);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'run1',
+          false
+        )
+      ).toEqual([runTableItems[0], runTableItems[2]]);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByRegex(
+          runTableItems,
+          'run2',
+          false
+        )
+      ).toEqual([runTableItems[1]]);
+    });
+  });
+
+  describe('getRenderableRuns', () => {
+    it('returns all runs associated with experiment', () => {
+      const exp1Result = selectors.TEST_ONLY.utils.getRenderableRuns(['exp1'])(
+        state
+      );
+      expect(exp1Result.length).toEqual(2);
+      expect(exp1Result[0].run).toEqual({...run1, experimentId: 'exp1'});
+      expect(exp1Result[1].run).toEqual({...run2, experimentId: 'exp1'});
+
+      const exp2Result = selectors.TEST_ONLY.utils.getRenderableRuns(['exp2'])(
+        state
+      );
+      expect(exp2Result.length).toEqual(3);
+      expect(exp2Result[0].run).toEqual({...run2, experimentId: 'exp2'});
+      expect(exp2Result[1].run).toEqual({...run3, experimentId: 'exp2'});
+      expect(exp2Result[2].run).toEqual({...run4, experimentId: 'exp2'});
+    });
+
+    it('returns two runs when a run is associated with multiple experiments', () => {
+      const result = selectors.TEST_ONLY.utils.getRenderableRuns([
+        'exp1',
+        'exp2',
+      ])(state);
+      expect(result.length).toEqual(5);
+      expect(result[0].run).toEqual({...run1, experimentId: 'exp1'});
+      expect(result[1].run).toEqual({...run2, experimentId: 'exp1'});
+      expect(result[2].run).toEqual({...run2, experimentId: 'exp2'});
+      expect(result[3].run).toEqual({...run3, experimentId: 'exp2'});
+      expect(result[4].run).toEqual({...run4, experimentId: 'exp2'});
+    });
+
+    it('returns empty list when no experiments are provided', () => {
+      expect(selectors.TEST_ONLY.utils.getRenderableRuns([])(state)).toEqual(
+        []
+      );
+    });
+  });
+
+  describe('filterRunItemsByHparamAndMetricFilter', () => {
```

Comment:
Can we add a test for filtering HParam AND Metric? With just one type will be enough imo.